src/Observable.ts

```diff
@@ -37,6 +37,13 @@ import {ErrorObservable} from './observable/throw';
 import {AjaxCreationMethod} from './observable/dom/ajax';
 import {WebSocketSubject} from './observable/dom/webSocket';
 
+import {CombineLatestSignature} from './operator/combineLatest';
+
+/* These types would be come standard types used through-out the library */
+export type ObservableOrPromise<T> = Observable<T> | Promise<T>;
```

Comment:
As long as we keep in mind further refactoring to use this common type in existing codebase either way's fine. but separate commit would be easier for you I assume? :)